app/controllers/bamposts_controller.rb

```diff
@@ -0,0 +1,70 @@
+class BampostsController < ApplicationController
+  before_action :set_bampost, only: %i[ show edit update destroy ]
+
+  # GET /bamposts or /bamposts.json
+  def index
+    @bamposts = Bampost.all
+  end
+
+  # GET /bamposts/1 or /bamposts/1.json
+  def show
+  end
+
+  # GET /bamposts/new
+  def new
+    @bampost = Bampost.new
+  end
+
+  # GET /bamposts/1/edit
+  def edit
+  end
+
+  # POST /bamposts or /bamposts.json
+  def create
+    @bampost = Bampost.new(bampost_params)
+    @bampost.image.attach(params[:image])
+
+    respond_to do |format|
+      if @bampost.save
+        format.html { redirect_to @bampost, notice: "Bampost was successfully created." }
+        format.json { render :show, status: :created, location: @bampost }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @bampost.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /bamposts/1 or /bamposts/1.json
+  def update
+    respond_to do |format|
+      if @bampost.update(bampost_params)
+        format.html { redirect_to @bampost, notice: "Bampost was successfully updated." }
+        format.json { render :show, status: :ok, location: @bampost }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @bampost.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /bamposts/1 or /bamposts/1.json
+  def destroy
+    @bampost.destroy
+    respond_to do |format|
+      format.html { redirect_to bamposts_url, notice: "Bampost was successfully destroyed." }
+      format.json { head :no_content }
+    end
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_bampost
+      @bampost = Bampost.find(params[:id])
+    end
+
+    # Only allow a list of trusted parameters through.
+    def bampost_params
+      params.require(:bampost).permit(:image, :content, :user_id)
+    end
+end
```

Comment:
Good job testing your controllers - have you tried testing your models,not much to test on that ground, but you could test validations like we did for our Acebook project. If you use  rspec you could use Shoulda Matcher, so it would be something like this: 
```
RSpec.describe Bampost, type: :model do
  describe 'validations' do

    it { should validate_presence_of(:content) }
  end
end
```
This[ article](https://github.com/thoughtbot/shoulda-matchers) has more on how to use Shoulda Matchers to test validations and more if you are interested